java-security/src/test/java/com/sap/cloud/security/token/validation/validators/JwtValidatorBuilderTest.java

```diff
@@ -110,4 +112,34 @@ public void buildWithAnotherValidator_containsAddedValidator() {
 				.contains(tokenValidator);
 	}
 
+	@Test
+	public void configureOtherServiceInstances() {
+		Collection clientIds = new ArrayList();
+		JwtAudienceValidator audienceValidator;
+
+		OAuth2ServiceConfiguration xsuaaConfig1 = xsuaaConfigBuilder.build();
+		OAuth2ServiceConfiguration xsuaaConfig2 = OAuth2ServiceConfigurationBuilder.forService(XSUAA)
+				.withClientId("sb-test-app2!b222").build();
+		OAuth2ServiceConfiguration xsuaaConfig3 = OAuth2ServiceConfigurationBuilder.forService(XSUAA)
+				.withClientId("sb-test-app3!b333").build();
+
+		clientIds.add(xsuaaConfig1.getClientId());
+		clientIds.add(xsuaaConfig2.getClientId());
+		clientIds.add(xsuaaConfig3.getClientId());
+
+		CombiningValidator<Token> combiningValidator = JwtValidatorBuilder
+				.getInstance(xsuaaConfig1)
+				.configureAnotherServiceInstance(xsuaaConfig2)
+				.configureAnotherServiceInstance(xsuaaConfig3)
+				.build();
+
+		for (Validator validator: combiningValidator.getValidators()) {
+			if(validator instanceof JwtAudienceValidator) {
+				assertThat(((JwtAudienceValidator) validator).clientIds).containsAll(clientIds);
+				return;
+			}
+		}
+		assertThat(true).isFalse(); // should never be called
```

Comment:
I would prefer to let the test fail with a message like `Assert.fail("No JwtAudienceValidator found that contains all clientIds!");`